Haskell/syntax_test_haskell.hs

```diff
@@ -186,3 +260,25 @@
     0XdeafBEEF42
 --  ^^^^^^^^^^^^ constant.numeric.integer.hexadecimal
 --  ^^ punctuation.definition.numeric.base
```

Comment:
Probably worth also adding a test for the `0x` prefix, just to ensure the case insensitivity is working.